server/src/app.ts

```diff
@@ -22,83 +24,170 @@ app.get("/alive", (req: Request, res: Response) => {
-  const id = uuidv4();
-
-  const newItem: Item = {
-    id,
-    title,
-    description,
-    dueDate,
-    createdAt: Math.floor(Date.now() / 1000),
-  };
-
-  items.set(id, newItem);
-  res.status(201).json(newItem);
-});
-//GET
-
-app.get("/api/v1/items/:id", (req: Request, res: Response) => {
-  const itemId: string = req.params.id;
-  const item: Item | undefined = items.get(itemId);
-
-  if (!item) {
-    return res.status(404).json({ message: "Resource not found" });
+async function startServer() {
+  try {
+    const db: Db = await connectToDb();
+
+    //POST
+
+    app.post("/api/v1/items", async (req: Request, res: Response) => {
+      try {
+        const {
+          title,
+          description,
+          categories,
+          ingredients,
+          steps,
+        }: {
+          title: string;
+          description: string;
+          categories: string[];
+          ingredients: string[];
+          steps: string[];
+        } = req.body;
+
+        const newItem: Item = {
+          title,
+          description,
+          categories,
+          ingredients,
+          steps,
+          createdAt: Math.floor(Date.now() / 1000),
+        };
+        const result: InsertOneResult<any> = await db
+          .collection("recipes")
+          .insertOne(newItem);
+        res
+          .status(200)
+          .json({ message: "recipe was added", postId: result.insertedId });
+      } catch (error) {
+        console.error("Error creating item:", error);
+        res.status(500).json({ error: "Failed to create item" });
+      }
+    });
+
+    //GET
+
+    app.get("/api/v1/items", async (req: Request, res: Response) => {
+      try {
+        const collection: Collection = db.collection("recipes");
+        const documents = await collection.find().toArray();
+
+        const recipes: Item[] = documents.map((recipe) => ({
+          title: recipe.title,
+          description: recipe.description,
+          categories: recipe.categories,
+          ingredients: recipe.ingredients,
+          steps: recipe.steps,
+          createdAt: recipe.createdAt,
+        }));
+
+        res.status(200).json(recipes);
+      } catch (error) {
+        console.error("Error getting items:", error);
+        res.status(500).json({ error: "Failed to get items" });
+      }
+    });
+
+    app.get("/api/v1/items/:id", async (req: Request, res: Response) => {
+      if (ObjectId.isValid(req.params.id)) {
+        try {
+          const collection: Collection = db.collection("recipes");
+          const document = await collection.findOne({
+            _id: new ObjectId(req.params.id),
+          });
+          res.status(200).json(document);
+        } catch (error) {
+          console.error("Error getting item:", error);
+          res.status(500).json({ error: "Failed to get items" });
+        }
+      } else {
+        res.status(500).json({ error: "Not valid document ID" });
```

Comment:
use 400 status here. If the recipe id is not valid that's not a server problem, that's a problem of a client who sent it - hence it's a `Bad Request`.